internal/compiler/typeloader.rs

```diff
@@ -36,12 +36,15 @@ pub struct ImportedName {
 }
 
 impl ImportedName {
-    pub fn extract_imported_names(
-        import: &syntax_nodes::ImportSpecifier,
-    ) -> Option<impl Iterator<Item = ImportedName>> {
-        import
-            .ImportIdentifierList()
-            .map(|import_identifiers| import_identifiers.ImportIdentifier().map(Self::from_node))
+    pub fn extract_imported_names(imports: &[syntax_nodes::ImportSpecifier]) -> Vec<ImportedName> {
```

Comment:
(nitpick:  You don't need to collect in a vector, you could just return an iterator.)

Reply:
I tried that, but the issue is that the returned iterator's lifetime includes the lifetime of `&import`, which can't be moved into the returned iterator, because it is borrowed by the call site while the iterator would also be alive :(

Reply:
you can add a `+ '_`  maybe that help.  Also a filter_map  followed by a flat_map would also help.
That said, don't spend to much time on that.

Reply:
Ah yes, thanks for pushing. The `filter_map` trick worked, as well as `+ '_` with a bit of gymnastics at the call site. I thought `import` was shared, but it was moved - that was the issue.